drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -265,6 +273,16 @@ static int _set_state(at86rf2xx_t *dev, netopt_state_t state)
             break;
         case NETOPT_STATE_RESET:
             at86rf2xx_hardware_reset(dev);
+            netdev_ieee802154_reset(&dev->netdev);
+            /* set short and long address */
+            at86rf2xx_set_addr_long(dev, (eui64_t *)dev->netdev.long_addr);
+            at86rf2xx_set_addr_short(dev, (network_uint16_t *)dev->netdev.short_addr);
+
+            if (!IS_ACTIVE(AT86RF2XX_BASIC_MODE)) {
+                static const netopt_enable_t enable = NETOPT_ENABLE;
+                netdev_ieee802154_set(&dev->netdev, NETOPT_ACK_REQ,
+                                      &enable, sizeof(enable));
+            }
             at86rf2xx_reset(dev);
```

Comment:
same here. I don't use these snippet anymore in #19015, this is just to ease the review of #19015. Should I address it anyway?
No, we don't seem to be using `NETOPT_STATE_RESET`, and I think as it is it will in most cases de-synchronize the radio with the MAC.